### PR TITLE
Update enclosure, facility, location, op batch, and user factories

### PR DESCRIPTION
**Pull request by todtb**
Continuation of #441 

### Description

Recent changes to the models and associations mean that it is now possible to make more precise and complete factories for the test suite.

This change introduces or updates Facility, Location, Operation Batch, and User factories and minor changes to the specs to incorporate said changes.
### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?
As these factories are used only in the test suite currently, rspec testing should be sufficient.

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -4,5 +4,8 @@
     password { "password" }
     password_confirmation { "password" }
     organization
+
+    trait(:as_admin)  { role { :admin } }
+    trait(:as_user)   { role { :user } }
```
Comment (benreyn) [inline spec/factories/users.rb:9]: I think id prefer `:user` just be the default, and only define a trait for admin. And Im not sure how I feel about the `as` prefix. `as_admin` reads to me like a macro or mixin we would use for a feature test, but for the factory I think `create(:user, :admin)` reads cleaner, IMO.

File: spec/features/locations/index_spec.rb [hunk 2]
```diff
@@ -8,9 +8,10 @@
   end
 
   it "Then I see a list of locations" do
-    locations = create_list(:location, 3, organization: user.organization)
+    facility = create(:facility)
+    locations = create_list(:location, 3, facility: facility, organization: user.organization)
 
-    visit facility_locations_path(locations.first.facility)
+    visit facility_locations_path(facility)
```
Comment (benreyn) [inline spec/features/locations/index_spec.rb:14]: :heart:

File: spec/factories/locations.rb [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
     facility
     organization
+
+    trait :with_enclosures do
+      transient { enclosures_count { 5 } }
+
+      after(:create) do |loc, eval|
+        create_list :enclosure, eval.enclosures_count, location: loc
+        loc.reload
+      end
```
Comment (benreyn) [inline spec/factories/locations.rb:13]: Its more verbose, but I think i prefer the full `location` here for clarity.

File: spec/factories/locations.rb [hunk 4]
```diff
@@ -1,7 +1,16 @@
 FactoryBot.define do
   factory :location do
-    name { "MyString" }
+    sequence(:name) { |n| "Location #{n}" }
     facility
     organization
+
+    trait :with_enclosures do
+      transient { enclosures_count { 5 } }
```
Comment (benreyn) [inline spec/factories/locations.rb:8]: I think `1` might be better defualt here. I worry that if we have to use this trait alot we'd end up creating alot of uneeded records, slowing down the test suit as a whole.

File: spec/factories/facility.rb [hunk 5]
```diff
@@ -1,8 +1,7 @@
 FactoryBot.define do
   factory :facility do
-    val = Faker::Name.unique.name
-    name { val }
-    code { val.gsub(/[aeiou|AEIOU|\s]+/, '').strip }
-    organization_id { FactoryBot.create(:organization).id }
+    name { Faker::Games::HalfLife.location.to_s }
```
Comment (benreyn) [inline spec/factories/facility.rb:3]: Making a note for myself to clean up usage of `Faker`. I dont think we really need it.

**Review by benreyn — COMMENTED** [review]
Thanks for tackling this @todtb! I left some notes. Let me know what you think. Also pinging @CraigJZ

File: spec/factories/users.rb (hunk 1 above)
Comment (todtb) [reply spec/factories/users.rb:9]: How about using `:user` as a default role but keeping the trait for consistency?

File: spec/factories/locations.rb (hunk 3 above)
Comment (todtb) [reply spec/factories/locations.rb:13]: I think I'll have a few changes in #445 then 😆

File: spec/factories/locations.rb (hunk 4 above)
Comment (todtb) [reply spec/factories/locations.rb:8]: Makes sense

File: spec/factories/facility.rb (hunk 5 above)
Comment (todtb) [reply spec/factories/facility.rb:3]: 😢  but fair enough

**Review by benreyn — APPROVED** [review]
Awesome! Thanks @todtb!

**Review by benreyn — APPROVED** [review]
Fixed some merge conflicts with #439. We may want to circle back if these create problems in the future

**Pull request merged**